src/main/java/MegaMillions.java

```diff
@@ -58,26 +55,45 @@ public void addMegaBalls(String line, List<Integer> megaBalls){
         }
     }
 
-    private List<Number> getUniqueNumbersWithOccurs(List<Integer> list) {
-        List<Number> numbersWithOccurs = new ArrayList<>();
-        for (int i = 1; i <= 10000; i++) {
+    private TreeMap<String, Integer> getUniqueNumbersWithOccurs(List<Integer> list) {
+        TreeMap<String, Integer> numbersWithOccurs = new TreeMap<>();
+        for (int i = 1; i <= Collections.max(list); i++) {
```

Comment:
why did you use collections.max? It doesnt really make sense to me. Collections.max will return highest number from the list but we want to go trough the whole list, so list.size() would be more appropriate.

Reply:
Ok, now I get what you've done but I think it can be done using one loop

Reply:
https://www.geeksforgeeks.org/count-occurrences-elements-list-java/